types/src/transaction/result.rs

```diff
@@ -251,10 +252,12 @@ impl ExecutionFinalResult {
     /// Because this function may panic, its use is generally discouraged. Instead, prefer
     /// to call into [`into_result`](ExecutionFinalResult::into_result) then pattern matching and handle the Err case explicitly.
     pub fn assert_success(self) -> ExecutionSuccess {
+        #[allow(clippy::unwrap_used)]
         self.into_result().unwrap()
     }
 
     pub fn assert_failure(self) -> ExecutionResult<TxExecutionError> {
```

Comment:
Just a random observation. I would expect the return value here to be `TxExecutionError`, not another "Result", but I did not look further into what ExecutionResult<> actually mean, so maybe it is justified, yet it is not intuitive from a quick glance, so might be misleading for some devs.